src/robot.js

```diff
@@ -352,7 +352,7 @@ class Robot {
     const full = path.join(filepath, path.basename(filename, ext))
 
     // see https://github.com/hubotio/hubot/issues/1355
-    if (!require.extensions[ext]) { // eslint-disable-line
+    if (['.js', '.mjs'].indexOf(ext) == -1) { // eslint-disable-line
```

Comment:
Does this have the effect of removing support for coffescript? I know there will be coffeescript based scripts out there both open source and private that are kicking around.

The reason we used this originally was that coffeescript and (I think) typescript would add to this. Looks like officially it's deprecated:

> In the past, this list has been used to load non-JavaScript modules into Node.js by compiling them on-demand. However, in practice, there are much better ways to do this, such as loading modules via some other Node.js program, or compiling them to JavaScript ahead of time.
>
>Avoid using require.extensions. Use could cause subtle bugs and resolving the extensions gets slower with each registered extension.

I think we need this for compatability for now, until there is a way to bundle your bot à la webpack or something similar. So, if we can disable eslint on this line, that would be preferred.

Reply:
What's wrong with just adding `.coffeescript` to the array?

Reply:
I've added `.coffee` to the array and it loaded a `.coffee` script.

Reply:
It's more that it's a hard-coded list, and there could be other extension types people register outside the context of hubot.

Based on my reading of the docs, the goal should be to transpile to javascript instead, so it becomes moot.